pcdsdevices/utils.py

```diff
@@ -528,14 +549,18 @@ def set_many(
     statuses = []
     log = owner.log if owner is not None else logger
     for signal, value in to_set.items():
+        # Set the actual signal value
         try:
-            st = signal.set(
-                value, timeout=timeout, settle_time=settle_time
-            )
+            try:
```

Comment:
This is definitely the simpler solution, but I was left wondering if we could inspect the signature and only pass the relevant kwargs through.  I think we could (`inspect.signature()` or `inspect.fullargspec()`), but there's no reason to over-complicate things here.

Reply:
The previous implementation used `inspect` but ran into issues